sdcm/utils/perftune_validator.py

```diff
@@ -0,0 +1,238 @@
+
+    def compare_option_file_yaml_with_temp_yaml_copy(self, option_file_dict) -> None:
+        temp_perftune_yaml_content_dict = self.executor.get_options_file_contents(use_temp_file=True)
+        if diff := DeepDiff(temp_perftune_yaml_content_dict, option_file_dict, ignore_order=True):
+            PerftuneResultEvent(
+                message=f"Mismatched results when comparing the output of the 'dump-options-file' command to "
+                        f"the content of the generated perftune.yaml file on {self.node}"
+                        f"\nActual result: '{temp_perftune_yaml_content_dict}'"
+                        f"\nExpected output: '{option_file_dict}'"
+                        f"\nDiff: {diff}",
+                severity=Severity.ERROR).publish()
+
+    @staticmethod
+    def _generate_new_irq_cpu_mask_string(mask_int_values) -> str:
+        new_masks = []
+        for mask_value in mask_int_values:
+            random_irq_int_value = random.randint(0, mask_value)
+            # The irq_cpu_mask cannot be greater than the cpu_mask, since it is a part of it
+            new_mask_string = "{0:x}".format(random_irq_int_value)  # converting back to base 16
+            padded_new_mask_string = f"0x{new_mask_string.rjust(8, '0')}"  # Padding
+            new_masks.append(padded_new_mask_string)
+        return ",".join(new_masks)
+
+    @staticmethod
+    def get_mask_int_values(complete_mask_string) -> list:
+        split_masks = complete_mask_string.split(",")
+        numerical_values = []
+        for mask_string in split_masks:
+            num_value = int(mask_string, base=16)
+            numerical_values.append(num_value)
+        return numerical_values
+
+    def compare_option_file_with_overridden_irq_cpu_mask_param(self, option_file_dict) -> None:
+        current_cpu_mask = option_file_dict["cpu_mask"]
+        cpu_mask_int_values = self.get_mask_int_values(current_cpu_mask)
+        random_irq_cpu_mask_string = self._generate_new_irq_cpu_mask_string(cpu_mask_int_values)
+        altered_option_file_contents = self.executor.get_options_file_contents(
+            override_irq_cpu_mask=random_irq_cpu_mask_string)
+        if altered_option_file_contents["irq_cpu_mask"] != random_irq_cpu_mask_string:
+            PerftuneResultEvent(
+                message=f"Despite overriding the irq_cpu_mask param, its value is was not altered properly in the "
+                        f"output of the 'dump-options-file' command on node {self.node}",
+                severity=Severity.ERROR).publish()
+
+    def compare_perftune_results(self) -> None:
+        PerftuneResultEvent(
+            message="Checking the output of perftune.py",
+            severity=Severity.NORMAL).publish()
+        try:
+            self.compare_cpu_mask()
+            self.compare_irq_cpu_mask()
+            if not self.node.is_nonroot_install:
+                # The following checks are not applicable for non-root installations,
```

Comment:
I don't know if it's intended or not, but non-root installs doesn't call perftune at all.